pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -368,11 +370,11 @@ func (w *worker) MachineDeployments() []extensionsv1alpha1.MachineDeployment {
 	return w.machineDeployments
 }
 
-func (w *worker) findExistingNodeTemplateByName(ctx context.Context, obj *extensionsv1alpha1.Worker, poolName string) *extensionsv1alpha1.NodeTemplate {
+func (w *worker) findExistingNodeTemplateByName(ctx context.Context, obj *extensionsv1alpha1.Worker, poolName string) (*extensionsv1alpha1.NodeTemplate, string) {
```

Comment:
```suggestion
func (w *worker) findNodeTemplateAndMachineTypeByPoolName(ctx context.Context, obj *extensionsv1alpha1.Worker, poolName string) (*extensionsv1alpha1.NodeTemplate, string) {
```

Reply:
addressed in [038182a](https://github.com/gardener/gardener/pull/5577/commits/038182a5a3b13bf21bda337cdf3bdc3612587e14)